lib/widgets/sample_viewer_page.dart

```diff
@@ -42,6 +44,35 @@ class _SampleViewerPageState extends State<SampleViewerPage> {
   var _ready = false;
   var _searchHasFocus = false;
 
+  final List<String> _searchPrefixes = [
+    'Try',
+    'Look for',
+    'Search',
+    'Explore',
+    'Type to explore',
+    'Check out',
+    'Discover',
+    'Start typing',
+    'Tap to search',
+  ];
+
+  final List<String> _questionPrefixes = [
+    'Need help with',
+    'Find out about',
+    'Interested in',
+    'Dig into',
+    'Want to learn',
+    'How about',
+    "Let's explore",
+  ];
+
+  // Limit keywords 50 characters.
+  final int _maxHintLength = 50;
+
+  List<String> _hintMessages = [];
+  int _currentHintIndex = 0;
```

Comment:
We want to use type inference as much as possible, so our pattern would be like this:
```dart
final _searchPrefixes = [ ... values ... ];
final _questionPrefixes = [ ... values ... ];
final _maxHintLength = 50;
final _hintMessages = <String>[];
var _currentHintIndex = 0;
```